docs/env.md

```diff
@@ -1,51 +1,147 @@
+}
+```
+
+Now when you run `npm run test`, inside of your Vue app, `process.env.API_KEY` will match the value of `API_KEY_TEST` in your `.env` file,
+
+When you run `npm run build`, it will match `API_KEY_PROD`. In this case it's the same value, because you want to run the `build` command with the same API_KEY as the `test` command, but they can be different in other environments, like when you develop locally:
+
+```
+// /.env
+API_URL_DEV=localhost:8080 // dev-server (proxying to a test-backend)
+API_URL_TEST=localhost:4001 // test-backend
+```
+```javascript
+// /config/variables/development.js
+module.exports = {
+  API_KEY: process.env.API_URL_DEV
+}
+// /config/variables/test.js
+module.exports = {
+  API_KEY: process.env.API_URL_TEST
+}
+```
+
+In your app, you can access `process.env.API_KEY` and will get the correct value in each mode.
+
+## A word on 'seperation of config from code' according to "The 12 Factor App"
+
+The 12 Factor App has a [section](https://12factor.net/config) about configuration values that recommends to extract *all* such values into environment variables, and never save them inside code - like we allow you do in the files mentioned above. 
+
+The main concerns are:
+
+1. storing sensitive information like API keys in code that's pushed to git can be insecure. Secrets may become public (i.e. over github), or they may leak between environments (i.e. a test environment accidentally using `production` credentials)
+2. Using config files for each build environment is inflexible. You will inevitably end up with a requirement for another environment that needs different config values than `test` or `production`, or `staging` and so on.
+
+We think this is mostly good advice, but in the context of this template, some things are to be considered to get some perspective:
+
+### Sensitive Information
+
+Every value that you pass into your app and use it can be accessed by your users one way or another. For that reason, frontend apps (the kind this template is meant for) should generally do not contain sensitive information like API secrets or database keys. If however you do need those values in your build setup, for example, it is absolutely good practice to pass those in via environment variables - jand don't pass them on to your Vue app through the variable files we discussed above.
+
+### Config files per environment are inflexible
+
+We agree to that as well. But `development`, `test` and `production` are not so much build environments as they are different *modes* that the application can run in. 
+
+Since there are many situations where you want to run more than one of those modes in the same environment, but with different values for a given variable (i.e. running `dev` and `test` locally, as further up), these different values have to be passed in with different environment variables. But these different variables have to be re-mapped to the same name if they ought to be accessible within the application:
```

Comment:
This flexibility is nice, but don't many people use multiple .env files (for different modes) to solve this problem?
Maybe worth adding a note about using separate .env files as the "switch" rather than remapping in the `config/variables` files.